eth-types/src/evm_types.rs

```diff
@@ -14,171 +14,113 @@ pub use opcode_ids::OpcodeId;
 pub use stack::{Stack, StackAddress};
 pub use storage::Storage;
 
-/// Wrapper type over `usize` which represents the program counter of the Evm.
-#[derive(Clone, Copy, Eq, PartialEq, Serialize, Deserialize, PartialOrd, Ord, Default)]
-pub struct ProgramCounter(pub usize);
-
-impl fmt::Debug for ProgramCounter {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_fmt(format_args!("0x{:06x}", self.0))
-    }
-}
-
-impl From<ProgramCounter> for usize {
-    fn from(addr: ProgramCounter) -> usize {
-        addr.0
-    }
-}
-
-impl From<usize> for ProgramCounter {
-    fn from(pc: usize) -> Self {
-        ProgramCounter(pc)
-    }
-}
-
-impl ProgramCounter {
-    /// Increase Self by one
-    pub fn inc(&mut self) {
-        self.0 += 1;
-    }
-
-    /// Increase Self by one and return the value before the increase.
-    pub fn inc_pre(&mut self) -> Self {
-        let pre = *self;
-        self.inc();
-        pre
-    }
-}
-
-/// Defines the gas left to perate.
-#[derive(Default, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
-pub struct Gas(pub u64);
-
-impl fmt::Display for Gas {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_fmt(format_args!("{}", self.0))
-    }
-}
-
-impl fmt::Debug for Gas {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_fmt(format_args!("{}", self.0))
-    }
-}
-
-impl From<Gas> for u64 {
-    fn from(value: Gas) -> Self {
-        value.0
-    }
-}
-
 /// Quotient for max refund of gas used
 pub const MAX_REFUND_QUOTIENT_OF_GAS_USED: usize = 5;
 /// Gas stipend when CALL or CALLCODE is attached with value.
 pub const GAS_STIPEND_CALL_WITH_VALUE: u64 = 2300;
 
 /// Defines the gas consumption.
-#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Default)]
-pub struct GasCost(pub u64);
+// #[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Default)]
+pub struct GasCost;
```

Comment:
It seems these two are no longer needed.
```suggestion
```

Reply:
We would still need the struct though (to implement it), am I wrong?

Reply:
The comment was removed.

Reply:
Yes, you are right. We still need to keep the GasCost struct.
I wasn't aware the constants are defined within the struct.